### ISSUE TITLE
Empty mysql endpoint results in unexpected EOF

### ISSUE THREAD

**Issue by joshgarnett** [hashicorp/terraform#7199]
terraform 0.6.16

```
provider "mysql" {
  endpoint = ""
  username = "user"
  password = "password"
}

resource "mysql_database" "db" {
  name = "somedb"
}
```

The above config will result in the following error
```
Error refreshing state: 1 error(s) occurred:

* unexpected EOF
```

Instead I'd expect it to error out that the endpoint is not set or that it was unable to connect to the provider.

**Comment by joshgarnett** [comment]
https://gist.github.com/joshgarnett/f8b62093032c51e73516e1a493c9a49f full crash log

**Comment by stack72** [comment]
Hi @joshgarnett 

Apologies for this. I have just opened a PR that add a little provider validation

```
% terraform apply
There are warnings and/or errors related to your configuration. Please
fix these before continuing.

Errors:

  * provider.mysql: Endpoint must not be an empty string
```

This should make the next release

Paul

**Comment by joshgarnett** [comment]
Thanks @stack72 !

**Comment by derFunk** [comment]
@stack72 It seems this still happens in 0.8.8 when the endpoint is referenced from another module, but not yet set up ("unkown" / "<computed>").

Example:
```
module "mysql" {
  source = "modules/mysql"

  # Input `db_address` to module mysql, which is using the mysql provider
  # in order to create databases on the db with endpoint `db_endpoint`.
  
  # Output `db_endpoint` from module rds, which created an aws_db_instance
  # and returns the endpoint of that instance.

  db_address = "${module.rds.db_endpoint}"
}
```

The log here says:
`2017/03/16 11:31:25 [DEBUG] Setting Unknown Variable Value for computed key: db_address`

and then:

```
Error refreshing state: 1 error(s) occurred:

* unexpected EOF
```

**Comment by derFunk** [comment]
But it's fixed in 0.9.0 it seems.

**Comment by stack72** [comment]
Hi @derFunk 
This specific issue has been "fixed" so it doesn't cause a panic BUT we have identified that chaining providers together:

i.e.

```
provider "aws" {}

provider "mysql" {
  parameter from aws resource
}
```

doesn't work as expected. Providers are evaluated at the same time and not in the dependency order

This is something we are looking at 

Thanks

Paul
